src/Transactions/UpdateContract.cs

```diff
@@ -0,0 +1,124 @@
+using Concordium.Sdk.Types;
+
+namespace Concordium.Sdk.Transactions;
+
+/// <summary>A deployment of a Wasm smart contract module.</summary>
+/// <param name="Amount">Amount of CCD to send to the instance.</param>
+/// <param name="Address">Address of the instance to update.</param>
+/// <param name="ReceiveName">Name of the receive function to call to update the instance.</param>
+/// <param name="Parameter">Name of the receive function to call to update the instance.</param>
+public sealed record UpdateContract(CcdAmount Amount, ContractAddress Address, ReceiveName ReceiveName, Parameter Parameter) : AccountTransactionPayload
+{
+    /// <summary>
+    /// Prepares the account transaction payload for signing.
+    /// </summary>
+    /// <param name="sender">Address of the sender of the transaction.</param>
+    /// <param name="sequenceNumber">Account sequence number to use for the transaction.</param>
+    /// <param name="expiry">Expiration time of the transaction.</param>
+    /// <param name="energy">
+    /// The amount of energy that can be used for contract execution.
+    /// The base energy amount for transaction verification will be added to this cost.
+    /// </param>
+    public PreparedAccountTransaction Prepare(
+        AccountAddress sender,
+        AccountSequenceNumber sequenceNumber,
+        Expiry expiry,
+        EnergyAmount energy
+    ) => new(sender, sequenceNumber, expiry, new(_baseCost.Value + energy.Value), this);
+
+    /// <summary>
+    /// The transaction specific cost for submitting this type of
+    /// transaction to the chain.
+    ///
+    /// This should reflect the transaction-specific costs defined here:
+    /// https://github.com/Concordium/concordium-base/blob/78f557b8b8c94773a25e4f86a1a92bc323ea2e3d/haskell-src/Concordium/Cost.hs
+    /// </summary>
+    private static readonly EnergyAmount _baseCost = new(300);
```

Comment:
I don't understand where this is coming from. There is no such thing for updating contracts.

Reply:
```rust
    /// Update a smart contract intance, giving it the given amount of energy
    /// for execution. The unique parameters are
    /// - `energy` -- the amount of energy that can be used for contract
    ///   execution. The base energy amount for transaction verification will be
    ///   added to this cost.
    pub fn update_contract(
        num_sigs: u32,
        sender: AccountAddress,
        nonce: Nonce,
        expiry: TransactionTime,
        payload: UpdateContractPayload,
        energy: Energy,
    ) -> PreAccountTransaction {
        let payload = Payload::Update { payload };
        make_transaction(
            sender,
            nonce,
            expiry,
            GivenEnergy::Add { num_sigs, energy },
            payload,
        )
    }
```

Reply:
This is from the base cost for updating instances, so the `updateContractInstanceBaseCost` in [Cost.hs](https://github.com/Concordium/concordium-base/blob/78f557b8b8c94773a25e4f86a1a92bc323ea2e3d/haskell-src/Concordium/Cost.hs).

Reply:
Yes, but I don't understand why this is added here. That cost is an internal implementation detail.

Whenever anybody supplies execution energy it will include this cost already. So we don't need to add it here.

Reply:
right... I'll remove this